wallet-new/src/Cardano/Wallet/WalletLayer/Kernel/Internal.hs

```diff
@@ -64,8 +68,24 @@ postponeUpdate :: MonadIO m => Kernel.PassiveWallet -> m ()
 postponeUpdate w = update' (w ^. Kernel.wallets) $ RemoveNextUpdate
 
 -- | Reset wallet state
-resetWalletState :: Kernel.PassiveWallet -> m ()
-resetWalletState = error "TODO: resetWaletState [CBR-393]"
+resetWalletState :: MonadIO m => Kernel.PassiveWallet -> m ()
+resetWalletState w = liftIO $ do
+    -- TODO: reset also the wallet worker (CBR-415)
+
+    -- stop restoration and empty it`s state.
+    Restore.stopAllRestorations w
```

Comment:
There is an issue here: a restoration might be started in between this stop and the modification of the DB. However, since this is used in testing only, I'd say, let's just document this potential problem and leave it.